src/Sulu/Component/Content/Types/Link.php

```diff
@@ -80,6 +80,13 @@ public function getViewData(PropertyInterface $property): array
             $result['rel'] = $value['rel'];
         }
 
+        if (null !== $value['provider'] && self::LINK_TYPE_EXTERNAL !== $value['provider'] && null !== $value['href']) {
+            $provider = $this->providerPool->getProvider($value['provider']);
+            $locale = $property->getStructure()->getLanguageCode();
+            $result['uuid'] = $value['href'];
+            $result['linkitems'] = $provider->preload([$value['href']], $locale, true);
+        }
+
```

Comment:
`view` should not resolve anything it should mostly just provide raw informations.

We should inside the `view` just return `href` and `provider` when that is the one you want to load more data and not recalling a resolver here:

```suggestion
        $result['href'] = $value['href'] ?? null;
```

Then it up to the user itself if the want to call something via a twig extension or not.